[WIP] dbt-materialize: add get clusters and schemas deploy macro (#29219)
MaterializeInc/materialize: doc/user/content/manage/dbt/development-workflows.md

<br>

1. By default, all clusters and schemas associated with models in your dbt
   project will be cloned as part of a blue/green deployment. To exclude specific
   clusters or schemas from the deployment, use the `deployment` variable in
   `dbt_project.yml`:

    ```yaml
    vars:
      deployment:
        default:
          exclude_clusters:
            # To specify multiple clusters, use [<cluster1_name>, <cluster2_name>].
            - <cluster_to_exclude>
          exclude_schemas:
            # To specify multiple schemas, use [<schema1_name>, <schema2_name>].
            - <schema_to_exclude>
    ```

>    If your dbt project includes [sinks](/manage/dbt/#sinks), you **must**
>    exclude the schema(s) and cluster(s) hosting these objects.

Reviewer: ```suggestion
   If your dbt project includes [sinks](/manage/dbt/#sinks), the schema(s) and
   cluster(s) hosting these objects are automatically excluded from cloning.
```

<br>

1. The `deploy_get_objects` macro automatically identifies the clusters and schemas in your dbt project. If you need to exclude specific clusters or schemas from the deployment, you can specify them in the `dbt_project.yml` file:

    ```yaml
    vars:
      deployment:
        default:
          exclude_clusters:
            # To specify multiple clusters, use [<cluster1_name>, <cluster2_name>].
            - <cluster_to_exclude>
          exclude_schemas:
            # To specify multiple schemas, use [<schema1_name>, <schema2_name>].
            - <schema_to_exclude>
    ```

   If you don't need to exclude any clusters or schemas, you can omit this configuration entirely.

1. To quickly view the clusters and schemas that will be included in the deployment, you can use the [`deploy_get_objects`](https://github.com/MaterializeInc/materialize/blob/main/misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_get_objects.sql) macro with the `debug` argument:

  ```bash
  dbt run-operation deploy_get_objects --args '{debug: true}'
  ```

>   This will output the list of clusters and schemas that will be included in the deployment, as well as any that have been excluded based on your configuration.

Reviewer 2: Tiny alignment issue. I believe these lines need to be shifted by 1 to the right as the current alignment causes a break in numbering.
<img width="529" alt="Screenshot 2024-09-11 at 2 07 26 PM" src="https://github.com/user-attachments/assets/56b67572-a975-4131-9ee3-087f21ba502c">

Author: Ah yes, good catch! Thanks @kay-kim
